library/ssl_tls.c

```diff
@@ -9177,8 +9177,9 @@ int mbedtls_ssl_conf_psk( mbedtls_ssl_config *conf,
     ssl_conf_remove_psk( conf );
 
     /* Check and set raw PSK */
-    if( psk == NULL || psk_len > MBEDTLS_PSK_MAX_LEN )
+    if( psk == NULL || psk_len == 0 || psk_len > MBEDTLS_PSK_MAX_LEN )
```

Comment:
Style: please avoid combining tests in the same `if` line when the then-block is a simple one-liner (like `return <some_constant>` or `goto cleanup`). Having them separate makes it easier to find out what went wrong when debugging, and it rarely makes any difference to the generated code.
```
    if( psk == NULL )
        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
    if( psk_len == 0 )
        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
    if( psk_len > MBEDTLS_PSK_MAX_LEN )
        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
```

Reply:
done